engine/frontend/src/component/log/Log.js

```diff
@@ -0,0 +1,178 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { VariableSizeList as List } from "react-window";
+import { useWindowResize } from "./useWindowResize";
+import { Flex, Box, Spacer } from "@chakra-ui/react";
+import { TriangleDownIcon, DownloadIcon } from '@chakra-ui/icons'
+import parse from 'html-react-parser';
+import hasAnsi from 'has-ansi';
+import stripAnsi from 'strip-ansi';
+import { useClipboard } from '@chakra-ui/react'
+import { saveTextAsFile } from "../../utils/download";
+
+const Convert = require('ansi-to-html');
+const convert = new Convert();
+
+const Row = ({ data, index, setSize, windowWidth}) => {
+  const rowRef = useRef();
+  
+  useEffect(() => {
+    if (rowRef.current !== undefined) {
+      setSize(index, rowRef.current.getBoundingClientRect().height);
+    } 
+  }, [setSize, index, windowWidth]);
+
+  if (data[index] !== undefined && data[index].length !== 0) {
+    let txt = data[index]
+    if (hasAnsi(data[index])) {
+      const parsedAnsi = convert.toHtml(data[index])
+      txt = parse(parsedAnsi)
+    }
+    return (
+        <div
```

Comment:
@Peeeekay ideally we wouldn't put in these `div`s but use chakra components instead. It'll be more techdebt to clean up.

Reply:
not sure why this is tech-debt? I am  not really using tailwind here; just `div` so we can keep this here for now. I don;t think we need to overkill this with chakra and react-window only accepts `inline-style`.. so additional apis provided by Chakra won't be useful.

Reply:
Yeah not noting use of tailwind, just not idiomatic use of Chakra. I would use Box, Container, etc. instead of div.